tests/core/test_farmer_harvester_rpc.py

```diff
@@ -170,37 +171,66 @@ async def have_plots():
+            update_interval_before = farmer_api.farmer.update_harvester_cache_interval
+            farmer_api.farmer.update_harvester_cache_interval = 1
+            farmer_api.farmer.harvester_cache = {}
+
             # Test farmer get_harvesters
             async def test_get_harvesters():
+                harvester.plot_manager.trigger_refresh()
+                await time_out_assert(5, harvester.plot_manager.needs_refresh, value=False)
                 farmer_res = await client.get_harvesters()
                 if len(list(farmer_res["harvesters"])) != 1:
+                    log.error(f"test_get_harvesters: invalid harvesters {list(farmer_res['harvesters'])}")
                     return False
                 if len(list(farmer_res["harvesters"][0]["plots"])) != num_plots:
+                    log.error(f"test_get_harvesters: invalid plots {list(farmer_res['harvesters'])}")
                     return False
                 return True
 
-            await time_out_assert(30, test_get_harvesters)
+            await time_out_assert_custom_interval(30, 1, test_get_harvesters)
+
+            # Reset cache and reset update interval to avoid hitting the rate limit
+            farmer_api.farmer.update_harvester_cache_interval = update_interval_before
+            farmer_api.farmer.harvester_cache = {}
 
             expected_result: PlotRefreshResult = PlotRefreshResult()
+            expected_result_matched = True
 
             def test_refresh_callback(refresh_result: PlotRefreshResult):
-                assert refresh_result.loaded_plots == expected_result.loaded_plots
-                assert refresh_result.removed_plots == expected_result.removed_plots
-                assert refresh_result.processed_files == expected_result.processed_files
-                assert refresh_result.remaining_files == expected_result.remaining_files
+                def test_value(name: str, actual: PlotRefreshResult, expected: PlotRefreshResult):
+                    nonlocal expected_result_matched
+                    try:
+                        actual_value = actual.__getattribute__(name)
+                        expected_value = expected.__getattribute__(name)
+                        if actual_value != expected_value:
+                            log.error(f"{name} invalid: actual {actual_value} expected {expected_value}")
+                            expected_result_matched = False
+                    except AttributeError as error:
+                        log.error(f"{error}")
+                        expected_result_matched = False
+
+                test_value("loaded_plots", refresh_result, expected_result)
+                test_value("removed_plots", refresh_result, expected_result)
+                test_value("processed_files", refresh_result, expected_result)
+                test_value("remaining_files", refresh_result, expected_result)
```

Comment:
the left hand side here seems better. It's shorter and easier to read and it actually fails the unit test if the values disagree. As far as I can tell, you just set `expected_result_matched`. which is then overwritten by the next test. afaict, if the `loaded_plots` field is a mismatch, and the remaining ones are matches, the test will still pass...

Reply:
Why would it still pass? It gets set to `True` before the refresh gets triggered [here](https://github.com/Chia-Network/chia-blockchain/pull/8491/files#diff-ebb107e316ce58dda857782dc62783bc29eb5c02e305e2f6c1ef44048f1b63feR229) and asserted after the the refresh [here](https://github.com/Chia-Network/chia-blockchain/pull/8491/files#diff-ebb107e316ce58dda857782dc62783bc29eb5c02e305e2f6c1ef44048f1b63feR233). `test_value` does not overwrite it, its `test_case` which resets it.

Reply:
And even though the previous version was shorter, see this https://github.com/Chia-Network/chia-blockchain/pull/8491/commits/bf1c6958733ab18f567c84aa215f28927b9e0292 commit's message for the reason for this change .

Reply:
I see, I misread the assignments to `=false`.